### ISSUE TITLE
Problems to resolve some artifacts in maven repo

### ISSUE THREAD

**Issue by wpc009** [coursier/coursier#436]
Some `scala` or `sbt` related projects have scalaVersion or sbtVersion in their module name. Which confuse the coursier resolver.
Like the following artifact.
```
https://repo1.maven.org/maven2/net/virtual-void/sbt-dependency-graph_2.10_0.13/0.8.0/sbt-dependency-graph-0.8.0.pom
```
When I try to use coursier to resolve it like `net.virtual-void:sbt-dependency-graph_2.10_0.13:0.8.0`. Coursier tries to reach `https://repo1.maven.org/maven2/net/virtual-void/sbt-dependency-graph_2.10_0.13/0.8.0/sbt-dependency-graph_2.10_0.13-0.8.0.pom` which does not exist. It seems that coursier thinks `sbt-dependency-graph_2.10_0.13` is the artifact's name, but the maven repo thinks `sbt-dependency-graph` is the actual name.
What's the workaround for this problem? Please.

**Comment by alexarchambault** [comment]
The workaround is two fold:
- `2.10` and `0.13` are attributes (in sbt, `addSbtPlugin` adds these attributes to dependencies), that should be passed to coursier like `net.virtual-void:sbt-dependency-graph;scalaVersion=2.10;sbtVersion=0.13:0.8.0` (from the command line, it is better to put this between quotes),
- for the directory name to be appended the `_2.10_0.13` suffix, the `--sbt-plugin-hack` option must be passed to coursier.

So the following works:
```
$ ./coursier fetch \
    "net.virtual-void:sbt-dependency-graph;scalaVersion=2.10;sbtVersion=0.13:0.8.0" \
    --sbt-plugin-hack
```

**Comment by wpc009** [comment]
I didn't see this `sbt-plugin-hack` thing in the README. Maybe it should be updated.
Thank you for the workaround and quickly response.

**Comment by wpc009** [comment]
I have another issue. I try to update a project's dependencies. But I got `coursier.ResolutionException: x not found` error. The artifact coursier complains not found is actually exists. I double checked. And I try to monitoring the network activity. coursier does not try to access the repository server when I run `update` job.

**Comment by wpc009** [comment]
I found a walk around of this problem. After hours searching on the google, somebody mentioned that coursier tries to download artifact's jar archive and sha1 hash in parallel, for some reason the sha1 is not corresponding to the jar archive. So, delete the artifact in the cache and re-try again, then coursier could resolve that artifact correctly.
